tests/rkt_volume_mount_test.go

```diff
@@ -0,0 +1,331 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build coreos kvm host fly
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"syscall"
+	"testing"
+
+	"github.com/appc/spec/schema"
+	"github.com/appc/spec/schema/types"
+	"github.com/coreos/rkt/tests/testutils"
+)
+
+// TODO: unite these tests with rkt_run_pod_manifest_test.go
+
+var (
+	boolTrue                             = true
+	boolFalse                            = false
+	tmpdir                               = createTempDirOrPanic("rkt-tests-fly")
+	tmpdirmountname         types.ACName = "dir1"
+	tmpdirpathpod                        = "/dir1"
+	tmpdir2path                          = path.Join(tmpdir, "dir2")
+	tmpdir2filepath                      = path.Join(tmpdir2path, "file")
+	tmpdir2filepathpod                   = "/dir1/dir2/file"
+	tmpdir2innerfilecontent              = "inner"
+	tmpdir2outerfilecontent              = "outer"
+)
+
+func prepareTmpDirWithRecursiveMountsAndFiles(t *testing.T) []func() {
+	cleanupFuncs := make([]func(), 0)
+
+	// create directory for /dir1
+	if err := os.MkdirAll(tmpdir2path, 0); err != nil {
+		t.Fatalf("Can't recreate inner temp directory %q: %v", tmpdir2path, err)
+	}
+	cleanupFuncs = append(cleanupFuncs, func() { os.RemoveAll(tmpdir) })
```

Comment:
In this scheme you're appending to `cleanupFuncs`, this means they'll get executed in the order they're appended if there's an error. Shouldn't you execute them in the inverse order?

You can either "prepend" here or change `executeFuncs()` to go through the list in reverse.